src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,171 @@ def test_set_beam_shift(self):
+            starting_time = time.time()
+            autostigmator_future = self.scanner.applyAutoStigmator(detector)
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            autostigmator_state = self.microscope.is_autostigmating(channel)
+            self.assertEqual(autostigmator_state, True)
+            self.assertIsInstance(autostigmator_future, ProgressiveFuture)
+
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            autostigmator_future.result(timeout=max_execution_time)
+
+            # Check if the time to perform the auto stigmation is not too long
+            self.assertLess(time.time() - starting_time, 30,
+                            "Execution of auto stigmation was stopped because it took more than %s seconds." %
+                            max_execution_time)
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+
+            stigmator_values_x.append(self.microscope.get_stigmator()[0])
+            stigmator_values_y.append(self.microscope.get_stigmator()[1])
+
+        # Check if all the stigmator values are within the required range each run
+        stigmator_range = self.microscope.stigmator_info()['range']
+        min_x_stig_values, max_x_stig_values = stigmator_range['x']
+        min_y_stig_values, max_y_stig_values = stigmator_range['y']
+        # TODO When auto stigmation works implement a check on the reproducibility of the auto stigmation values and
+        #  define the range with which they may differ from each other. In theory similar values should be returned.
+
+        # Test if an error is raised when an invalid detector name is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoStigmator("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(channel)
+        self.assertEqual(autostigmator_state, False)  # Check if state remained unchanged
+
+    def test_apply_auto_contrast_brightness(self):
+        """
+        Test for the auto contrast brightness functionality.
+        """
+        # Check all the different detector types by looping over them
+        for role, channel in DETECTOR2CHANNELNAME.items():
+            # Start auto contrast brightness and check if it is running.
+            auto_contrast_brightness_future = self.scanner.applyAutoContrastBrightness(role)
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(channel)
+            self.assertEqual(auto_contrast_brightness_state, True)
+            self.assertIsInstance(auto_contrast_brightness_future, ProgressiveFuture)
+            # Give microscope/simulator to properly start before canceling, to prevent an error.
+            time.sleep(2.5)
```

Comment:
Without sleeping the error:
`Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/sonic/development/odemis/src/odemis/driver/test/xt_client_test.py", line 421, in test_apply_auto_contrast_brightness
    auto_contrast_brightness_future.cancel()
  File "/home/sonic/development/odemis/src/odemis/model/_futures.py", line 311, in cancel
    if not (canceller and canceller(self)):
  File "/home/sonic/development/odemis/src/odemis/driver/xt_client.py", line 982, in _cancelAutoContrastBrightness
    self.parent.set_auto_contrast_brightness(future._channel_name, XT_STOP)
  File "/home/sonic/development/odemis/src/odemis/driver/xt_client.py", line 516, in set_auto_contrast_brightness
    self.server.set_auto_contrast_brightness(name, state)
  File "/home/sonic/.local/lib/python3.6/site-packages/Pyro5/client.py", line 476, in __call__
    return self.__send(self.__name, args, kwargs)
  File "/home/sonic/.local/lib/python3.6/site-packages/Pyro5/client.py", line 243, in _pyroInvoke
    raise data  # if you see this in your traceback, you should probably inspect the remote traceback as well
OSError: [Errno -2147219405] Set auto contrast brightness state failed. Error message: None`

Is raised, having this waiting time here fixing this.